tests/importer/test_importer.py

```diff
@@ -84,29 +83,24 @@ def test_import_error_cleanup():
 
 @pytest.mark.skipif(sys.dont_write_bytecode,
                     reason="Bytecode generation is suppressed")
-def test_import_autocompiles():
+def test_import_autocompiles(tmp_path):
     "Test that (import) byte-compiles the module."
 
-    with tempfile.NamedTemporaryFile(suffix='.hy', delete=True) as f:
-        f.write(b'(defn pyctest [s] (+ "X" s "Y"))')
-        f.flush()
+    p = tmp_path / 'mymodule.hy'
+    p.write_text('(defn pyctest [s] (+ "X" s "Y"))')
 
-        pyc_path = importlib.util.cache_from_source(f.name)
-
-        try:
-            os.remove(pyc_path)
-        except OSError:
-            pass
-
-        spec = importlib.util.spec_from_file_location('mymodule', f.name)
+    def ip(path):
```

Comment:
This method name is meaningless. Please change it something more immediately clear.

Reply:
I'm always surprised when people want much more descriptive names from local subroutines than just-as-local temporary variables. I guess I still have some habits from programming in Haskell over a decade ago. But I'll change it, yes.

Reply:
Other people need to be able to work on and understand this code. It doesn't need to have the same the same explicitness as a top level public method `imppath` or the like would be fine

Reply:
But that would imply that people than me exist, and their needs even matter… are you sure about that?

Reply:
a wild departure from your usual solipsism i know